The code cannot run under Windows.

Hi Team,

Could you please optimize the code? the current code cannot run under Windows, and the error message is as follows: 

C:\gotestwaf-master\gotestwaf-master>go run ./cmd --url=http://39.100.84.77 --reportFormat=html --noEmailReport --skipWAFIdentification
# github.com/wallarm/gotestwaf/internal/scanner
internal\scanner\scanner.go:345:36: undefined: syscall.SIGUSR1
![01](https://github.com/wallarm/gotestwaf/assets/168075326/040ca527-d473-4fa7-8f48-75a361bf31eb)
![01](https://github.com/wallarm/gotestwaf/assets/168075326/b9cd6980-6e1a-4c4b-9eb6-8d741f342f80)

--- a/internal/scanner/scanner.go
+++ b/internal/scanner/scanner.go
@@ -10,8 +10,6 @@ import (
 	"math/rand"
 	"net/http"
 	"net/url"
-	"os"
-	"os/signal"
 	"regexp"
 	"strings"
 	"sync"
@@ -321,32 +319,11 @@ func (s *Scanner) Run(ctx context.Context) error {
 	// separately.
 	var requestsCounter uint64
 
-	userSignal := make(chan os.Signal, 1)
-	signal.Notify(userSignal, syscall.SIGUSR1)
-	defer func() {
-		signal.Stop(userSignal)
-		close(userSignal)
-	}()
-
-	go func() {
-		for {
-			select {
-			case _, ok := <-userSignal:
-				if !ok {
-					return
-				}
-
-				s.logger.
-					WithFields(logrus.Fields{
-						"sent":  atomic.LoadUint64(&requestsCounter),
-						"total": s.db.NumberOfTests,
-					}).Info("Testing status")
-
-			case <-ctx.Done():
-				return
-			}
-		}
-	}()
+	closeListener, err := s.testStatusSignalHandler(ctx, &requestsCounter)
+	if err != nil {
+		return err
+	}
+	defer closeListener()
 
 	for e := 0; e < gn; e++ {
 		go func() {

--- a/internal/scanner/scanner_signal_handler_unix.go
+++ b/internal/scanner/scanner_signal_handler_unix.go
@@ -1,0 +1,41 @@
+package scanner
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"sync/atomic"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+func (s *Scanner) testStatusSignalHandler(ctx context.Context, requestsCounter *uint64) (func(), error) {
+	userSignal := make(chan os.Signal, 1)
+	signal.Notify(userSignal, syscall.SIGUSR1)
+
+	go func() {
+		for {
+			select {
+			case _, ok := <-userSignal:
+				if !ok {
+					return
+				}
+
+				s.logger.
+					WithFields(logrus.Fields{
+						"sent":  atomic.LoadUint64(requestsCounter),
+						"total": s.db.NumberOfTests,
+					}).Info("Testing status")
+
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return func() {
+		signal.Stop(userSignal)
+		close(userSignal)
+	}, nil
+}

--- a/internal/scanner/scanner_signal_handler_windows.go
+++ b/internal/scanner/scanner_signal_handler_windows.go
@@ -1,0 +1,9 @@
+package scanner
+
+import (
+	"context"
+)
+
+func (s *Scanner) testStatusSignalHandler(_ context.Context, _ *uint64) (func(), error) {
+	return func() {}, nil
+}

